### PR TITLE
Red xenobio paint now works on vehicles

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Causing them to go faster.

:cl: Kor
rscadd: Red xenobio potions now work on vehicles (janitor cart, ATV, secway, etc), causing them to go faster.
/:cl:

### CODE REVIEW

**Comment by GunHog** [comment]
https://www.youtube.com/watch?v=NMv3-ZJeoYI&feature=youtu.be&t=19

**Comment by PKPenguin321** [comment]
Vroom

File: code/modules/research/xenobiology/xenobiology.dm [hunk 1]
```diff
@@ -282,17 +282,28 @@
...
 		return
-	if(C.slowdown <= 0)
-		user << "<span class='warning'>The [C] can't be made any faster!</span>"
-		return..()
+	if(istype(C, /obj/item))
+		var/obj/item/I = C
+		if(I.slowdown <= 0)
+			user << "<span class='warning'>The [C] can't be made any faster!</span>"
+			return..()
+		user <<"<span class='notice'>You slather the red gunk over the [C], making it faster.</span>"
```
Comment (KazeEspada) [inline code/modules/research/xenobiology/xenobiology.dm]: this will make a double message

**Comment by Lumipharon** [comment]
ayyyyy

**Comment by Anonmare** [comment]
Red wunz go fasta

**Comment by neersighted** [comment]
Flawless :+1:

**Comment by optimumtact** [comment]
what stops me reapplying the paint over and over until the vehicle reaches ludicrous speed?

**Comment by KorPhaeron** [comment]
@optimumtact 

It sets it to the max speed in one application anyway

**Comment by optimumtact** [comment]
I see
I didn't know what I was thinking

File: code/modules/research/xenobiology/xenobiology.dm [hunk 2]
```diff
@@ -282,17 +282,27 @@
...
+		user << "<span class='warning'>The potion can only be used on items or vehicles!</span>"
 		return
-	if(C.slowdown <= 0)
-		user << "<span class='warning'>The [C] can't be made any faster!</span>"
-		return..()
+	if(istype(C, /obj/item))
+		var/obj/item/I = C
+		if(I.slowdown <= 0)
+			user << "<span class='warning'>The [C] can't be made any faster!</span>"
+			return..()
```
Comment (tkdrg) [inline code/modules/research/xenobiology/xenobiology.dm]: Why is this allowed
Just `return ..()` like a sane person

File: code/modules/research/xenobiology/xenobiology.dm [hunk 3]
```diff
@@ -282,17 +282,27 @@
...
+		I.slowdown = 0
+
+	if(istype(C, /obj/vehicle))
+		var/obj/vehicle/V = C
+		if(V.vehicle_move_delay <= 0)
+			user << "<span class='warning'>The [C] can't be made any faster!</span>"
+			return..()
```
Comment (tkdrg) [inline code/modules/research/xenobiology/xenobiology.dm]: same as above

**Comment by tkdrg** [comment]
I could see this getting annoying, but we can always add a timeout if we must.

**Pull request merged**